asm-to-pil/src/vm_to_constrained.rs

```diff
@@ -454,6 +456,86 @@ impl<T: FieldElement> ASMPILConverter<T> {
         res
     }
 
```

Comment:
Could you please document this function? It's quite large and takes `&mut self`

Reply:
should be better now after restructuring the code